### ISSUE TITLE
Wrong usage of syscall open in some shellcraft templates

### ISSUE THREAD

**Issue by david942j** [Gallopsled/pwntools#1102]
Some templates invoke `SYS_open` in a wrong way,
such as:
https://github.com/Gallopsled/pwntools/blob/a208251d5f70202aae4d884aa955f0972b3c2621/pwnlib/shellcraft/templates/amd64/linux/cat.asm#L11
The position of `oflag` and `mode` are reversed.
It should be `${syscall('SYS_open', 'rsp', 'O_RDONLY', 0)}`

Other wrong usages can be found in i386 and aarch64 (not familiar with aarch64, but it seems wrong)

This bug doesn't produce wrong shellcode because the value of `O_RDONLY` is zero,
but the output of shellcraft might confuse users, for example:
```python
context.arch = 'i386'
print shellcraft.cat('flag')
#    /* push 'flag\x00' */
#    ...
#    /* open(file='esp', oflag=0, mode='O_RDONLY') */   <--- wrong
#    mov ebx, esp
#    ...
#    int 0x80
```

**Comment by zachriggle** [comment]
Thanks for pointing this out! Would you mind submitting a Pull Request that fixes this?

**Comment by david942j** [comment]
OK, I'll do it

**Comment by david942j** [comment]
closed by #1106
